server/test/endpoint_roles.test.js

```diff
@@ -0,0 +1,104 @@
+const { describe, it } = require('mocha')
+const fetch = require('node-fetch')
+const assert = require('node:assert')
+const { getAdmin1IdPToken } = require('./lib/tokens')
+const { getServerUrl } = require('./lib/urls')
+const dbAdminRole = require('database/layer/admin-role')
+
+describe('Admin Roles API Tests', () => {
+  const url = getServerUrl()
+  let requestHeaders
+  before(async function () {
+    // GET IDENTITY-PROVIDER TOKEN
+    const userAccessToken = await getAdmin1IdPToken()
+    requestHeaders = {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${userAccessToken}`
+    }
+  })
+
+  describe('Create Roles', () => {
+    const clientId = 'test-client1'
+    const roleName = 'Role for E2E Test'
+
+    it('Should create a new role', async () => {
+      const postBody = {
+        client_id: clientId,
+        name: roleName,
+        description: 'the desc value'
+      }
+
+      const res = await fetch(`${url}/roles`, {
+        method: 'POST',
+        headers: requestHeaders,
+        body: JSON.stringify(postBody)
+      })
+
+      assert.strictEqual(res.status, 201)
+      const role = await res.json()
+      assert.ok(typeof role.key === 'number')
+      assert.ok(typeof role.clientkey === 'number')
+      assert.strictEqual(role.name, roleName)
+    })
+    it('Should create two roles with same name for different clients', async () => {
+      const postBody = {
+        client_id: clientId,
+        name: roleName,
+        description: 'the desc value'
+      }
+
+      const res = await fetch(`${url}/roles`, {
+        method: 'POST',
+        headers: requestHeaders,
+        body: JSON.stringify(postBody)
+      })
+      postBody.client_id = 'test-client2'
+      const res2 = await fetch(`${url}/roles`, {
+        method: 'POST',
+        headers: requestHeaders,
+        body: JSON.stringify(postBody)
+      })
+
+      // delete the second Role
+      await dbAdminRole.deleteRoleByClientRolename('test-client2', roleName)
```

Comment:
I see that we are not deleting both roles, as the next test is dependent on it. Shouldn't we ensure that each test is fully independent to avoid a chain failure if one test fails or to prevent deadwoods in the database?

Reply:
@k1ch  the "first" role is being deleted by the `afterEach` function down around line 100. I know it's a bit weird to manually delete one item in the test, and then rely on an `afterEach` to delete the other. Maybe I will just add a comment to make this clear.